druid/src/widget/painter.rs

```diff
@@ -107,6 +107,19 @@ impl<T> Painter<T> {
 }
 
 impl<T: Data> BackgroundBrush<T> {
+    /// Request paint if the BackgroundBrush changed.
+    pub fn update(&mut self, ctx: &mut UpdateCtx, old_data: &T, data: &T, env: &Env) {
+        match self {
+            Self::ColorKey(key) => {
+                if ctx.env_key_changed(key) {
+                    ctx.request_paint();
+                }
+            }
```

Comment:
Not a blocker, but my personal preference is for inlining the conditional, like:

```suggestion
            Self::ColorKey(key) if ctx.env_key_changed(key) => {
                ctx.request_paint();
            }
```